### DISCUSSION TITLE
Add pnpm to default setup

### DISCUSSION THREAD

**Discussion by Jolg42** [community/community#5056] [Codespaces] [answered]
npm and yarn package managers are provided by default.

pnpm is an awesome package manager which is used by many people/companies as of today.
See https://pnpm.io/users - https://github.com/pnpm/pnpm

Currently opening any project using pnpm as a package and then trying to run `pnpm i` results in

`bash: pnpm: command not found`

Making pnpm available by default would make the experience better so there is no need to install pnpm for each new Codespace.

Current workaround:
https://pnpm.io/installation

**Comment by tanmayeekamath** [comment, marked as answer]
Hey @Jolg42, thanks for your feedback. I will capture this. 

The [default image](https://aka.ms/ghcs-default-image) is quite heavy in the sense that it has a variety of things installed in it mainly to enable a catch all scenario for users just getting started with Codespaces. While we try to include a wide set of components, we are aware that there might be things missing. 

To customize your Codespaces configuration based on your requirements and scenarios, we support creating a custom configuration for your repo using the devcontainer file instead of using the default image. 
Here is a doc on how to get started with [creating your own devcontainer config for your repo](https://docs.github.com/en/codespaces/customizing-your-codespace/configuring-codespaces-for-your-project).

Please reach out if you have any questions :)

**Reply by Jolg42** [reply]
Thanks for the detailed answer, I will look into the configuration then!

Hopefully pnpm will be added one day 💚

**Reply by Jolg42** [reply]
Here is the config I crafted https://github.com/prisma/prisma/tree/main/.devcontainer

The Dockerfile for pnpm
```diff
# See here for image contents: https://github.com/microsoft/vscode-dev-containers/tree/v0.187.0/containers/typescript-node/.devcontainer/base.Dockerfile

# [Choice] Node.js version: 16, 14, 12
ARG VARIANT="14-buster"
FROM mcr.microsoft.com/vscode/devcontainers/typescript-node:0-${VARIANT}

# [Optional] Uncomment this section to install additional OS packages.
# RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
#     && apt-get -y install --no-install-recommends <your-package-list-here>

# [Optional] Uncomment if you want to install an additional version of node using nvm
# ARG EXTRA_NODE_VERSION=10
# RUN su node -c "source /usr/local/share/nvm/nvm.sh && nvm install ${EXTRA_NODE_VERSION}"

+ # To install more global node packages
+ RUN su node -c "npm install -g pnpm ts-node"
```
